lib/ditcher.js

```diff
@@ -45,13 +46,83 @@ var Ditcher = function (cfg, lgr, versionNumber, callback) {
 
 util.inherits(Ditcher, EventEmitter);
 
+/**
+ * Creates an object containing the expression key and value queried.
+ *
+ * @param {String} expression - This is the operator used in query.
+ * @param {ObjectId|Number|Boolean} value - This is the value being quried.
+ *
+ * @returns {Object} expressionQuery - contains the expressions key and query value.
+ */
+function expressionQuery(expression, value) {
+  var expressionQuery = {};
+  expressionQuery[expression] = value;
+  return expressionQuery[expression];
+}
+
+// contains functions to convert the query value based on the query type.
+var convertors = {
+  "Date": function (value) {
+    return new Date(value);
+  },
+  "ObjectId": function (value) {
+    return ObjectId(value);
+  },
+  "Any": function (value, field, expression) {
+    var arr = [];
+    var query = {};
+    var str = value;
+    query[field] = str;
+    arr.push(query);
+    try {
+      query[field] = expression ? expressionQuery(expression, value) : ObjectId(value);
+      arr.push(query);
+    } catch (err) { }
```

Comment:
Yes, if an error occurs here it means it could not cast the value to an `ObjectId`. No need to do anything else with this